### DISCUSSION TITLE
Extract content of bullet list out

### DISCUSSION THREAD

**Discussion by jirijakes** [jgm/pandoc#9532] [Q&A] [answered]
I have this markdown which causes problems during further processing and I guess is even invalid:

```md
# Hello

- ### One
  - Abc
- ### Two
  - Def
```

I would like to remove the outer bullets and turn them into simple headers using a Lua filter. So I would like to get this:

```md
# Hello

### One
  - Abc
### Two
  - Def
```

I managed to get to the right point but I fail to return the right thing. IIUC, if I manage to return list of headers, it would replace the bullet list.

```lua
function BulletList(list)
   if list.content[1][1].t == 'Header' and list.content[1][1].level == 3 then
      return list.content
   end
end
```

The `list.content` is exatly what I would like to replace the bullet list with. The step I am missing is how to return it in a correct way.

```haskell
List {
  [Header 3 ("one",[],[]) [Str "One"],BulletList [[Plain [Str "Abc"]]]],
  [Header 3 ("two",[],[]) [Str "Two"],BulletList [[Plain [Str "Def"]]]]
}
```

- `return list.content` fails with “Block, list of Blocks, or compatible element expected, got List”
- `return table.unpack(list.content)` returns only first header
- `return list.content[1]` or `return list.content[2]` return the one header but I cannot figure out how to return them all

Thank you.

**Comment by jgm** [comment]
I'm actually not sure!  @tarleb should be able to answer this.

**Comment by tarleb** [comment, marked as answer]
This is what I came up with:

``` lua
local function flatten (list)
  local result = {}
  for i, item in ipairs(list) do
    for j, block in ipairs(item) do
      result[#result + 1] = block
    end
  end
  return result
end

function BulletList(list)
  if list.content[1][1].t == 'Header' and list.content[1][1].level == 3 then
    return flatten(list.content)
  end
end
```

Does it do what you need?

**Reply by jirijakes** [reply]
Yes, that's it.

Thank you, @tarleb.
